prediction_market_agent/agents/microchain_agent/tools.py

```diff
@@ -0,0 +1,53 @@
+from eth_typing import HexStr, HexAddress, ChecksumAddress
+from prediction_market_agent_tooling.markets.agent_market import (
+    AgentMarket,
+    FilterBy,
+    SortBy,
+)
+from prediction_market_agent_tooling.markets.omen.omen import OmenAgentMarket
+from prediction_market_agent_tooling.markets.omen.omen_contracts import OmenConditionalTokenContract, \
+    OmenFixedProductMarketMakerContract
+from prediction_market_agent_tooling.markets.omen.omen_subgraph_handler import OmenSubgraphHandler
+from web3 import Web3
+from web3.types import Wei
+
+
+def address_to_checksum_address(address: str) -> ChecksumAddress:
+    return ChecksumAddress(HexAddress(HexStr(address)))
```

Comment:
This will change the typing so mypy will be happy, but in order to actually checksum the address, it's needed to call

```
Web3.to_checksum_address(self.collateralTokenAddress)
```

(if typing is changed to checksum address, but the address isn't actual checksummed, transactions can fail)

Reply:
And `to_checksum_address` is typed correctly on the library, so no need fur the wrapper function